src/as2fm/scxml_converter/data_types/type_utils.py

```diff
@@ -211,8 +211,5 @@ def recurse_on_array(
 
             return recurse_on_array(data_value, array_info.array_dimensions, expected_types)
         else:
-            # Small hack to accept integer values in case type is float
-            if expected_data_type is float:
-                expected_types = (int, float)
```

Comment:
Hacks that remove other small hacks that seemed weird are the best types of hacks. :)